invenio_oauthclient/contrib/cern.py

```diff
@@ -504,6 +505,7 @@ def on_identity_changed(sender, identity):
     :param identity: The user identity where information are stored.
     """
     if isinstance(identity, AnonymousIdentity):
+        disconnect_identity(identity)
```

Comment:
IMO this is "too late" to clean the global identity - it is already the AnonymousIdentity so it should have no leftovers at this stage. Can't we connect to logout signal somehow and clean it there?

Reply:
I think this is good thinking here.

These signals are from flask principal, while flask-login provides the explicit [signals](https://flask-login.readthedocs.io/en/latest/#signals) login, logout etc.. you refer to.

`invenio-accounts` comes with a [listener](https://github.com/inveniosoftware/invenio-accounts/blob/b156d3bcabba2b2c4008b4c023a36d44f87888e7/invenio_accounts/sessions.py#L93) that is cleaning some stuff in the session. I believe that those who declare some props in the session should also be responsible for cleaning them up, i.e. `cern_resource` value. The `identity.provides` are part of the principal module, thus their `reset` in that signal.

I do not have a strong opinion on it, maybe @zzacharo can advice on this.

Reply:
I don't think this is too late. This signal is fired when the identify changes, from Authenticated -> Anonymous, so it should be correct.
Let's discuss IRL if needed.